artifactory/commands/transferfiles/transfer_test.go

```diff
@@ -0,0 +1,162 @@
+package transferfiles
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/transfer"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+const uuidTokenForTest = "af14706e-e0c1-4b7d-8791-6a18bd1fd339"
+
+type validateVersionTestSuite struct {
+	testName      string
+	curVersion    string
+	errorExpected bool
+}
+
+func TestValidateDataTransferPluginMinimumVersion(t *testing.T) {
+	var pluginVersion string
+	testServer, serverDetails, _ := transfer.CreateRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/"+pluginsExecuteRestApi+"dataTransferVersion" {
+			content, err := json.Marshal(utils.VersionResponse{Version: pluginVersion})
+			assert.NoError(t, err)
+			_, err = w.Write(content)
+			assert.NoError(t, err)
+		}
+	})
+	defer testServer.Close()
+
+	srcPluginManager, err := createSrcRtUserPluginServiceManager(serverDetails)
```

Comment:
This code is repeated for each test method. Let's give the tests equal respect as the production code.
I suggest exporting it to a function. (we don't really need the "return", because we always look at the first assertion anyway)